tests/integration/activity_test.go

```diff
@@ -68,58 +73,70 @@ func TestActivity_Starring(t *testing.T) {
+	_, err := client.Activity.DeleteRepositorySubscription(owner, repo)
 	if err != nil {
-		t.Fatalf("Activity.ListWatchers returned error: %v", err)
-	}
-
-	if len(watchers) == 0 {
-		t.Errorf("Activity.ListWatchers('google', 'go-github') returned no watchers")
-	}
-
-	// the rest of the tests requires auth
-	if !checkAuth("TestActivity_Watching") {
-		return
+		t.Fatalf("Activity.DeleteRepositorySubscription returned error: %v", err)
 	}
 
-	// first, check if already watching google/go-github
-	sub, _, err := client.Activity.GetRepositorySubscription("google", "go-github")
+	// check again and verify not watching
+	sub, _, err := client.Activity.GetRepositorySubscription(owner, repo)
 	if err != nil {
 		t.Fatalf("Activity.GetRepositorySubscription returned error: %v", err)
 	}
 	if sub != nil {
-		t.Fatalf("Already watching google/go-github.  Please manually stop watching it first.")
+		t.Fatalf("Still watching google/go-github after deleting subscription.")
 	}
+}
 
+func createSubscription(t *testing.T) {
 	// watch google/go-github
-	sub = &github.Subscription{Subscribed: github.Bool(true)}
-	_, _, err = client.Activity.SetRepositorySubscription("google", "go-github", sub)
+	sub := &github.Subscription{Subscribed: github.Bool(true)}
+	_, _, err := client.Activity.SetRepositorySubscription(owner, repo, sub)
 	if err != nil {
 		t.Fatalf("Activity.SetRepositorySubscription returned error: %v", err)
 	}
 
 	// check again and verify watching
-	sub, _, err = client.Activity.GetRepositorySubscription("google", "go-github")
+	sub, _, err = client.Activity.GetRepositorySubscription(owner, repo)
 	if err != nil {
 		t.Fatalf("Activity.GetRepositorySubscription returned error: %v", err)
 	}
 	if sub == nil || !*sub.Subscribed {
 		t.Fatalf("Not watching google/go-github after setting subscription.")
 	}
+}
 
-	// delete subscription
-	_, err = client.Activity.DeleteRepositorySubscription("google", "go-github")
+func TestActivity_Watching(t *testing.T) {
+	watchers, _, err := client.Activity.ListWatchers(owner, repo, nil)
 	if err != nil {
-		t.Fatalf("Activity.DeleteRepositorySubscription returned error: %v", err)
+		t.Fatalf("Activity.ListWatchers returned error: %v", err)
 	}
 
-	// check again and verify not watching
-	sub, _, err = client.Activity.GetRepositorySubscription("google", "go-github")
+	if len(watchers) == 0 {
+		t.Errorf("Activity.ListWatchers('google', 'go-github') returned no watchers")
+	}
+
+	// the rest of the tests requires auth
+	if !checkAuth("TestActivity_Watching") {
+		return
+	}
+
+	// first, check if already watching google/go-github
+	sub, _, err := client.Activity.GetRepositorySubscription(owner, repo)
 	if err != nil {
 		t.Fatalf("Activity.GetRepositorySubscription returned error: %v", err)
 	}
+
+	// If already subscribing, delete then recreate subscription.
 	if sub != nil {
-		t.Fatalf("Still watching google/go-github after deleting subscription.")
+		deleteSubscription(t)
+		createSubscription(t)
+		return
```

Comment:
Really minor style thought for your consideration.

When it comes to error handling, I really like to [indent error flow](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow). That way, the "normal" path is at normal indentation, and the more unusual error handling is indented.

Here, I think using this "early return" style is not as nice, because the two cases are equal in my books, neither is an error, and neither is less likely than the other. They're just 2 branching paths, either you're already watching or not.

So I think it's slightly nicer to express that the two paths are essentially equal by not using early return, but instead... Personally, I'd use my favorite `switch`:

```Go
switch {
case sub != nil: // If already subscribing, delete then recreate subscription.
	deleteSubscription(t)
	createSubscription(t)
case sub == nil: // Otherwise, create subscription and then delete it.
	createSubscription(t)
	deleteSubscription(t)
}
```

But if you're not as big a fan of `switch` as me and [Andrew](https://twitter.com/enneff/status/575126465236332545), you could just `if ... else ...`.

Current code is absolutely fine, I'm just being nitpicky and looking for ways to make it even better.